src/System.Diagnostics.Process/tests/ProcessTests.cs

```diff
@@ -1875,31 +1875,41 @@ public void TestLongProcessIsWorking()
             Assert.True(p.HasExited);
         }
 
-        [PlatformSpecific(TestPlatforms.AnyUnix)]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "The archive of test project with SupplementalTestData in csproj, does not include the additional test data")]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.Uap, "Retrieving information about local processes is not supported on uap")]
         [ActiveIssue(37054, TestPlatforms.OSX)]
         [Fact]
         public void GetProcesses_LongProcessName()
         {
-            string commandName = "sleep";
+            string originalProcessName = "SleepTestAppNative";
+            string longProcessName = $"{originalProcessName}_file_with_quite_a_long_name";
+            string assemblyDirectory = Path.GetDirectoryName(typeof(ProcessTests).Assembly.Location);
+
+            string originalProcessPath = Path.Combine(assemblyDirectory, originalProcessName);
+            string longProcessPath = Path.Combine(TestDirectory, longProcessName);
 
-            // sleep program doesn't exist on some flavor
-            // in flavors such as Alpine, sleep is a busybox command which cannot be renamed
-            if (!IsProgramInstalled(commandName) || IsProgramInstalled("busybox"))
+            if (PlatformDetection.IsWindows)
             {
-                return;
+                File.Copy(originalProcessPath + ".exe", longProcessPath + ".exe");
             }
+            else
+            {
+                File.Copy(originalProcessName, longProcessPath);
 
-            string longProcessName = "123456789012345678901234567890";
-            string sleepCommandPathFileName = Path.Combine(TestDirectory, longProcessName);
-            File.Copy(GetProgramPath(commandName), sleepCommandPathFileName);
+                // set file mode to executable, becuase it loses the executable flag when extracted (on Helix) from zip archive;
+                // it may not be required once ZipFile APIs support Unix permissions: https://github.com/dotnet/corefx/issues/17342
+                using (var process = Process.Start(new ProcessStartInfo { FileName = "chmod", Arguments = $@"755 ""{longProcessPath}""" }))
+                {
+                    process.WaitForExit();
+                }
+            }
 
             // start sleep program and wait for some seconds
-            using (Process px = Process.Start(new ProcessStartInfo { FileName = sleepCommandPathFileName , Arguments = "30", UseShellExecute = true}))
+            using (Process px = Process.Start(new ProcessStartInfo { FileName = longProcessPath, Arguments = "30" }))
```

Comment:
It occurs to me, is there any reason this is not 30 and not infinite (or huge). Since we're going to kill it very quickly anyway. I can't imagine why 30 wouldn't be enough but I have been surprised several times given how many zillions of times these tests get run and the machines are busy at the same time.

Reply:
Sure, will update. There is no reason why it can't use upper cross-platform limit of `4294967` defined in `program.c` file.